kinto/plugins/accounts/authentication.py

```diff
@@ -15,8 +30,12 @@ def account_check(username, password, request):
 
     hashed = existing['password'].encode(encoding='utf-8')
     pwd_str = password.encode(encoding='utf-8')
-    if hashed == bcrypt.hashpw(pwd_str, hashed):
-        return True  # Match! Return anything but None.
+    # Check if password is valid (it is a very expensive computation)
+    if bcrypt.checkpw(pwd_str, hashed):  # Match!
+        # Remember result so we don't have to calculate bcrypt hash next time.
+        cache.set(cache_key, "1", ttl=30)  # Time to live: 30 seconds
```

Comment:
I think it would be great to expose the TTL in settings.

Reply:
Can you think of a use case for that?

Reply:
Can you think of a use case for that?

Reply:
One can increase this value to have more cache hits (e.g across periodic requests). Refreshing the TTL at each request would also be a great feature IMO, but this can be done in another PR.

Reply:
Thought about it too, but couldn't figure out how to implement cache that would expire after one request. Any ideas?